core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -499,6 +499,10 @@ public void process(@Nullable MappingMetaData mappingMd, String concreteIndex) {
             }
         }
 
+        if ("".equals(id)) {
```

Comment:
Out of curious, should we do `Strings.hasText(id) == false` here, to disallow the id `" "` also?